fem/coefficient.hpp

```diff
@@ -806,12 +941,42 @@ class VectorSumCoefficient : public VectorCoefficient
 class ScalarVectorProductCoefficient : public VectorCoefficient
 {
 private:
+   double aConst;
    Coefficient * a;
    VectorCoefficient * b;
 
 public:
+   ScalarVectorProductCoefficient(double A, VectorCoefficient &B);
    ScalarVectorProductCoefficient(Coefficient &A, VectorCoefficient &B);
 
+   void SetAConst(double A) { a = NULL; aConst = A; }
+   double GetAConst() const { return aConst; }
+
+   void SetACoef(Coefficient &A) { a = &A; }
+   Coefficient * GetACoef() const { return a; }
+
+   void SetBCoef(VectorCoefficient &B) { b = &B; }
+   VectorCoefficient * GetBCoef() const { return b; }
+
+   virtual void Eval(Vector &V, ElementTransformation &T,
+                     const IntegrationPoint &ip);
+   using VectorCoefficient::Eval;
+};
+
+/// Vector coefficient defined as a normalized vector field (returns v/|v|)
+class NormalizedVectorCoefficient : public VectorCoefficient
+{
+private:
+   VectorCoefficient * a;
+
+   double tol;
+
+public:
+   NormalizedVectorCoefficient(VectorCoefficient &A, double tol = 1e-6);
```

Comment:
Please add a description for the tolerance parameter.

Reply:
Yes, this would be helpful.